assets/js/stripe.js

```diff
@@ -398,305 +508,281 @@ jQuery( function( $ ) {
+		 */
 		onSubmit: function( e ) {
 			if ( ! wc_stripe_form.isStripeChosen() ) {
 				return;
 			}
 
-			if ( ! wc_stripe_form.isStripeSaveCardChosen() && ! wc_stripe_form.hasSource() && ! wc_stripe_form.hasToken() && ! wc_stripe_form.hasIntent() ) {
-				e.preventDefault();
+			/**
+			 * ToDo: The logic here was restructured a bit, make sure it still works well.
+			 */
 
-				wc_stripe_form.block();
+			// If a source, token, or an intent is already in place, submit the form as usual.
+			if ( wc_stripe_form.isStripeSaveCardChosen() || wc_stripe_form.hasSource() || wc_stripe_form.hasToken() || wc_stripe_form.hasIntent() ) {
+				return true;
+			}
 
-				// Stripe Checkout.
-				if ( 'yes' === wc_stripe_params.is_stripe_checkout && wc_stripe_form.isStripeModalNeeded() && wc_stripe_form.isStripeCardChosen() ) {
-					if ( 'yes' === wc_stripe_params.is_checkout ) {
-						return true;
-					} else {
-						wc_stripe_form.openModal();
-						return false;
-					}
+			// Open the Stripe Checkout modal.
+			if ( 'yes' === wc_stripe_params.is_stripe_checkout && wc_stripe_form.isStripeModalNeeded() && wc_stripe_form.isStripeCardChosen() ) {
+				if ( 'yes' === wc_stripe_params.is_checkout ) {
+					return true;
+				} else {
+					wc_stripe_form.openModal();
+					return false;
 				}
+			}
 
-				/*
-				 * For methods that needs redirect, we will create the
-				 * source server side so we can obtain the order ID.
-				 */
-				if (
-					wc_stripe_form.isBancontactChosen() ||
-					wc_stripe_form.isGiropayChosen() ||
-					wc_stripe_form.isIdealChosen() ||
-					wc_stripe_form.isAlipayChosen() ||
-					wc_stripe_form.isSofortChosen() ||
-					wc_stripe_form.isP24Chosen() ||
-					wc_stripe_form.isEpsChosen() ||
-					wc_stripe_form.isMultibancoChosen()
-				) {
-					if ( $( 'form#order_review' ).length ) {
-						$( 'form#order_review' )
-							.off(
-								'submit',
-								this.onSubmit
-							);
-
-						wc_stripe_form.form.submit();
-
-						return false;
-					}
-
-					if ( $( 'form.woocommerce-checkout' ).length ) {
-						return true;
-					}
-
-					if ( $( 'form#add_payment_method' ).length ) {
-						$( 'form#add_payment_method' )
-							.off(
-								'submit',
-								this.onSubmit
-							);
+			// For methods that needs redirect, we will create the source server side so we can obtain the order ID.
+			if (
+				wc_stripe_form.isBancontactChosen() ||
+				wc_stripe_form.isGiropayChosen() ||
+				wc_stripe_form.isIdealChosen() ||
+				wc_stripe_form.isAlipayChosen() ||
+				wc_stripe_form.isSofortChosen() ||
+				wc_stripe_form.isP24Chosen() ||
+				wc_stripe_form.isEpsChosen() ||
+				wc_stripe_form.isMultibancoChosen()
+			) {
+				if ( $( 'form#order_review' ).length ) {
+					$( 'form#order_review' )
+						.off(
+							'submit',
+							this.onSubmit
+						);
 
-						wc_stripe_form.form.submit();
+					wc_stripe_form.form.submit();
 
-						return false;
-					}
+					return false;
```

Comment:
Changing this to `return true;` will submit the form normally, so there should be no need to un-attach the `onSubmit` handler or calling `submit()` again.

In fact, all this logic above and below can be rewritten as just:
```php
if ( wc_stripe_form.isAlternatePersonality() ) {
    return true;
}
```

But please double-check me on that.

Reply:
Double-checked. 👍